### ISSUE TITLE
[Bug] Comment Multiline Input Field state isn't getting reset once the comment has been Send

### ISSUE THREAD

**Issue by Nabhag8848** [RocketChat/Apps.Notion#41]
### What happened?

- Comment Multiline Input Field state isn't getting reset once the comment has been Send. it should reset the field to empty string.

### Steps to reproduce

1. run '/notion comment'
2. select page and comment on page.
3. see the comment input field.

### Relevant ScreenShots

https://github.com/RocketChat/Apps.Notion/assets/65061890/07322c17-3fdf-43cc-b837-a6fdf1c5f6ea

### Version

v6.5

### What browsers are you seeing the problem on?

Chrome

**Comment by brf153** [comment]
@Nabhag8848 capabilities of my integration are
![Screenshot from 2024-01-22 21-48-31](https://github.com/RocketChat/Apps.Notion/assets/107868772/c8a1d5bb-e443-49d5-8025-9cc3a3de4a78)

Still, I am not being able to make comments on a page.
![Screenshot from 2024-01-22 21-49-45](https://github.com/RocketChat/Apps.Notion/assets/107868772/6d61ca5d-5b83-4b99-b728-c8dcf5720c4f)

**Comment by Nabhag8848** [comment]
> @Nabhag8848 capabilities of my integration are
> ![Screenshot from 2024-01-22 21-48-31](https://github.com/RocketChat/Apps.Notion/assets/107868772/c8a1d5bb-e443-49d5-8025-9cc3a3de4a78)
> 
> Still, I am not being able to make comments on a page.
> ![Screenshot from 2024-01-22 21-49-45](https://github.com/RocketChat/Apps.Notion/assets/107868772/6d61ca5d-5b83-4b99-b728-c8dcf5720c4f)
> 

connect again maybe you didn't have capabilities when you connected

**Comment by brf153** [comment]
You mean that I should refresh the page or should I use the command of `rc-apps deploy url username password` again?

**Comment by brf153** [comment]
Yeah I am working on the pagination one. I have made the recursive functions. I just need to check whether they are working fine or not. This issue seems easier than that one so I thought of making a pr for this one first.

**Comment by Nabhag8848** [comment]
> Yeah I am working on the pagination one. I have made the recursive functions. I just need to check whether they are working fine or not. This issue seems easier than that one so I thought of making a pr for this one first.

In my opinion, prioritise that first and come up with this later, this will make things easier for us for reviewing.

**Comment by brf153** [comment]
@Nabhag8848 oki

**Comment by brf153** [comment]
I will finish the previous one first

**Comment by VipinDevelops** [comment]
Hello, @Nabhag8848, I'm interested in working on this issue.

**Comment by brf153** [comment]
@VipinDevelops I was actually working on this issue

**Comment by Sayan4444** [comment]
Hi @Nabhag8848 when we click submit,the input text should be equals=''. This feature is already implemented but its only clearing the input field value from the server and not from the client i.e. after sending a message you cant send the same message again as you cant send a emtpy message. I demonstrated it in the below video.

https://github.com/RocketChat/Apps.Notion/assets/112304873/07802bdc-74f1-450d-9df4-1c4ebd2a34a4

If there is a function or a way in which we can also update the client side when the its server side value is changed i.e. we force the input field controlling it from the server, then it should work

**Comment by Nabhag8848** [comment]
> Hi @Nabhag8848 when we click submit,the input text should be equals=''. This feature is already implemented but its only clearing the input field value from the server and not from the client i.e. after sending a message you cant send the same message again as you cant send a emtpy message. I demonstrated it in the below video.
> 
>  2024-01-29.00-36-53.mp4 
> If there is a function or a way in which we can also update the client side when the its server side value is changed i.e. we force the input field controlling it from the server, then it should work

Yep. you got the point. #43 this issue is affecting this too. we have viewErrors everywhere but currently for some reason its not working. `handleCommentInputAction` is the function which is having the logic where we are storing the state on dispatch action of keyboard click when the input comment is written. try to get where this is called and most probably the state isn't resetting when comment is made.

**Comment by Sayan4444** [comment]
handleCommentInputAction is called for every keystrokes we make but we want to this delete the input field text for submit. Moreover input field state can be reset anywhere using this

await modalInteraction.clearInputElementState(
                CommentPage.COMMENT_INPUT_ACTION
            );

https://github.com/RocketChat/Apps.Notion/assets/112304873/8f01a180-fced-4628-9f86-37c619bc76bb

I hope my explaination is good enough and do tell me if I lack any understanding

**Comment by Nabhag8848** [comment]
> @Nabhag8848 handleCommentInputAction is called for every keystrokes we make but we want to this delete the input field text for submit. Moreover input field state can be reset anywhere using this
> 
> await modalInteraction.clearInputElementState( CommentPage.COMMENT_INPUT_ACTION );
> 
>  2024-01-29.01-19-30.mp4 
> I hope my explaination is good enough and do tell me if I lack any understanding

Yep but i believe we are clearlying the input on action of comment, issue lies somewhere else.
